.github/workflows/style_checks.yaml

```diff
@@ -0,0 +1,33 @@
+name: Style
+
+on: [push, pull_request]
+
+jobs:
+  check_clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly
+          components: clippy
+          override: true
+    - uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all-features
+
+  check_fmt:
+    name: Rust-fmt
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly
+          components: rustfmt
+          override: true
```

Comment:
Why does this use the nightly toolchain? I try to stick to stable as much as I can.

Reply:
(Sometimes it's needed for fancy features, I just don't know in this case)

Reply:
If I run rustfmt on stable it complains that stable rustfmt is deprecated and instructions me to use nightly. I think perhaps the newer versions of rustfmt require features in nightly. This is not our code on nightly but rather the latest rustfmt is supposed to be on nightly.

Reply:
Okay, fair enough. I wish they'd be making _opposite_ progress (deprecating nightly and working to get it working on stable) but I don't mind quite so much for dev tooling.